onnxruntime/core/optimizer/qdq_transformer/qdq_propagation.cc

```diff
@@ -19,7 +19,7 @@ static bool CanNodePropagate(const Node& node) {
 }
 
 static bool TryCancelOutDQQPair(Graph& graph, Node& dq_node, Node& q_node) {
-  if (!QDQ::IsQDQPairSupported(graph, q_node, dq_node)) {
+  if (!QDQ::IsQDQPairSupported(GraphViewer(graph), q_node, dq_node)) {
     return false;
```

Comment:
Creating a GraphViewer is somewhat expensive. Can we pass through the GraphViewer that was created in QDQPropagationTransformer::ApplyImpl?

Reply:
It's kind of hard to pass GraphViewer through the call stacks, it's also here,
https://github.com/microsoft/onnxruntime/blob/a4f25b7ae1e7c7898960312d99172100b51cca46/onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc#L82

One way is we can try to use const GraphViewer for NodeSelector completely, and get rid of Select(Graph& graph,...)
Please see this commit of the PR https://github.com/microsoft/onnxruntime/pull/9621/commits/f4d1c6bdbaec7a2a909aefaec1cd48bc188632da

But I did not come up with a good class to represent Selection,

Reply:
"One way is we can try to use const GraphViewer for NodeSelector completely, and get rid of Select(Graph& graph,...)"

sounds good. conceptually, it makes sense to me for node selection to not modify the graph (and require a `Graph&`). perhaps consider updating `NodesToOptimize` to not use mutable `Node*`s?